hotosm/tasking-manager
# Add similar projects at the bottom of the project detail page.

closes #5626
This PR adds similar projects at the bottom of the project detail page.
![realted_projects](https://user-images.githubusercontent.com/67958673/226308328-57da000c-ff2e-4725-9234-cad942f8d5c9.png)


**Implementation**:
- Similarity between the projects and given project will be calculated using cosine similarity and top 4 projects will be returned by default.
- Similarity between projects is calculated using these fields. -> `default_locale, difficulty, mapping_types, country, categories`
- All published projects except private ones are displayed  on non logged in view while only projects in which user has permission to map are shown for a logged in user.
- The function to create project matrix, which is the most significant part of calculating project similarity will be cached for a day. 

**How to test:**
- Install backend dependencies as new dependencies has been added in this PR.
- Start frontend and backend locally.
- Go to project detail page of any published projects
- Check if Similar projects button is visible on the footer alongside question & comments, contributors ....
- Click on similar projects button. You should be scrolled to Similar projects section.
- On the similar projects section 4 other similar projects should be displayed.
